CRM/Event/Form/ManageEvent/TabHeader.php

```diff
@@ -199,9 +199,11 @@ public static function process(&$form) {
           $action = 'browse';
         }
 
-        $tabs[$key]['link'] = CRM_Utils_System::url("civicrm/event/manage/{$key}",
-          "{$reset}action={$action}&id={$eventID}&component=event{$tabs[$key]['qfKey']}"
-        );
+        $link = "civicrm/event/manage/{$key}";
+        $query = "{$reset}action={$action}&id={$eventID}&component=event{$tabs[$key]['qfKey']}";
+
+        $tabs[$key]['link'] = (isset($value['link']) ? self::url($value['link']) :
+          CRM_Utils_System::url($link, $query));
```

Comment:
It makes sense to check for `$value['link']` and use that (when provided.)

I'm not so sure about filtering the link through  `CRM_Event_Form_ManageEvent_TabHeader::url()`. I guess the problem being solved is that, when composing a link from the event tabs to an internal Angular page, the link needs to meet some constraints (ie using a certain 'snippet' and moving the fragment into a param). 

However, there's no guarantee that link for the tab points to a Civi page. I believe the intent of the tab hooks is to also support third-party pages (eg Drupal/Joomla/WordPress pages). In those cases, you might also need to manipulate the URL, but the manipulations would be different. (e.g. instead of "?route=xyz" they might say "?fragment=xyz", and "?snippet" might have a conflicted different meaning).

Separately, the need to construct the URL this way is not particular to the `ManageEvent` form. If you embedded a Civi-Angular page in some other tabset, it would need to do similar URL manipulation.

Maybe `CRM_Event_Form_ManageEvent_TabHeader::url()` should be called `CRM_Utils_System::snippetUrl()`, and a hook implementer would write:

```php
function hook_civicrm_tabset(...) {
  //...
  $tab['foo']['link'] = CRM_Utils_System::snippetUrl(...) 
  //...
}
```

Then, in `CRM_Event_Form_ManageEvent`, there's no extra filtering -- just pass `$value['link']` directly through.